packages/block-editor/src/components/link-control/text-highlight.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { escapeRegExp } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Fragment,
+} from '@wordpress/element';
+
+const TextHighlight = ( { text = '', highlight = '' } ) => {
```

Comment:
This is a cool component. Not necessarily something we'd address in this PR, but I wonder if it's something to consider moving to `@wordpress/components` in the future. 

Adding this to other autocompleters might also be an idea as a follow-up to this PR.

Reply:
I thought the same thing

Reply:
Ok so I'll remove it from this PR and then create a new PR to add the component to `@wordpress/components`?

Then we can re-add it once that component exists?

Reply:
As long as it's not exported publicly (which I don't think it is), I think it's fine to keep it here for now, and then propose moving it to components when this PR is merged. That way it won't be a blocker for this PR.

Reply:
Ok done https://github.com/WordPress/gutenberg/issues/18074